scripts/checks.sh

```diff
@@ -46,7 +49,105 @@ function log_status()
 	jq -cn --argjson g "${1}" --arg f "${2}" --arg s "${3}" '[{"name":$f,"success":$g,"status":$s}]'
 }
 
+function test_upstream_dns()
+{
+	# force dnsmasq to print statistics to journal to read them back
+	systemctl kill -s USR1 dnsmasq
```

Comment:
Is this safe to do?

Reply:
@shaunmulligan `USR1` is a custom signal that can be sent, `dnsmasq` responds to this signal by dumping statistics to its log.

Reply:
That's correct, it's not as scary as it seems :) we've done that in the diagnostics logs for a while now: https://github.com/balena-io/device-diagnostics/blob/master/scripts/diagnose.sh#L101